File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/service/UserService.java
package ru.tggc.telegrambotcore.service;

import ru.tggc.telegrambotcore.dto.ChatDto;
import ru.tggc.telegrambotcore.dto.UserDto;
import ru.tggc.telegrambotcore.dto.UserRole;

public interface UserService {

    boolean checkRoles(Long id, UserRole[] requiredRoles);

    void saveOrUpdate(UserDto dto, ChatDto chatDto);
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/service/UserService.kt
package ru.tggc.telegrambotcore.service

import ru.tggc.telegrambotcore.dto.ChatDto
import ru.tggc.telegrambotcore.dto.UserDto
import ru.tggc.telegrambotcore.dto.UserRole

/**
 * Сервис для работы с БД пользователей
 */
interface UserService {
    /**
     * Метод для проверки ролей пользователя
     *
     * @param id id пользователя в телеграм
     */
    fun checkRoles(id: Long, requiredRoles: Array<UserRole>): Boolean

    /**
     * Метод сохранения или обновления пользователя в БД
     */
    fun saveOrUpdate(dto: UserDto, chatDto: ChatDto)
}
